tests/integration/client/feedback/dataset/test_filtered.py

```diff
@@ -0,0 +1,89 @@
+
+from tests.factories import DatasetFactory, RecordFactory, TextFieldFactory, TextQuestionFactory, UserFactory
+
+
+@pytest.mark.asyncio
+class TestFilteredRemoteFeedbackDataset:
+    @pytest.mark.parametrize("role", [UserRole.owner, UserRole.admin])
+    @pytest.mark.parametrize(
+        "status",
+        [
+            FeedbackResponseStatusFilter.draft,
+            FeedbackResponseStatusFilter.missing,
+            FeedbackResponseStatusFilter.discarded,
+            FeedbackResponseStatusFilter.submitted,
+            [FeedbackResponseStatusFilter.discarded, FeedbackResponseStatusFilter.submitted],
+        ],
+    )
+    async def test_filter_by(
+        self, role: UserRole, status: Union[FeedbackResponseStatusFilter, List[FeedbackResponseStatusFilter]]
+    ) -> None:
+        dataset = await DatasetFactory.create()
+        await TextFieldFactory.create(dataset=dataset, required=True)
+        await TextQuestionFactory.create(dataset=dataset, required=True)
+        await RecordFactory.create_batch(dataset=dataset, size=10)
+        user = await UserFactory.create(role=role, workspaces=[dataset.workspace])
+
+        api.init(api_key=user.api_key)
+        remote_dataset = FeedbackDataset.from_argilla(id=dataset.id)
+        filtered_dataset = remote_dataset.filter_by(response_status=status)
+        assert isinstance(filtered_dataset, FilteredRemoteFeedbackDataset)
+        assert isinstance(filtered_dataset.records, FilteredRemoteFeedbackRecords)
+        assert all([isinstance(record, RemoteFeedbackRecord) for record in filtered_dataset.records])
+
+    @pytest.mark.parametrize("role", [UserRole.owner, UserRole.admin])
+    @pytest.mark.parametrize(
+        "status",
+        [
+            FeedbackResponseStatusFilter.draft,
+            FeedbackResponseStatusFilter.missing,
+            FeedbackResponseStatusFilter.discarded,
+            FeedbackResponseStatusFilter.submitted,
+            [FeedbackResponseStatusFilter.discarded, FeedbackResponseStatusFilter.submitted],
+        ],
+    )
+    async def test_not_implemented_methods(
+        self, role: UserRole, status: Union[FeedbackResponseStatusFilter, List[FeedbackResponseStatusFilter]]
+    ) -> None:
+        dataset = await DatasetFactory.create()
+        text_field = await TextFieldFactory.create(dataset=dataset, required=True)
+        await TextQuestionFactory.create(dataset=dataset, required=True)
+        await RecordFactory.create_batch(dataset=dataset, size=10)
+        user = await UserFactory.create(role=role, workspaces=[dataset.workspace])
+
+        api.init(api_key=user.api_key)
+        remote_dataset = FeedbackDataset.from_argilla(id=dataset.id)
+        filtered_dataset = remote_dataset.filter_by(response_status=status)
+        assert isinstance(filtered_dataset, FilteredRemoteFeedbackDataset)
+
+        with pytest.raises(NotImplementedError):
+            filtered_dataset.delete_records(remote_dataset.records[0])
```

Comment:
we should check also the shortcutted method `filtered_dataset.records.delete`

Reply:
Yes, I'll add some other `NotImplemented` exceptions for those